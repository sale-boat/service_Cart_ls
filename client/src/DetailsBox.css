.details{
    color :#0066c0;
 }
 
 .details:hover{
   color :  #B12704;
   cursor : pointer;
 }


 .tooltip{
     position: relative; 
     display : inline-block;
 }

.exitButton1{
    width: 30px;
    height: 30px;
    border-radius: 3px;
}
 
 .message{
     position : absolute;
     border : 1px solid #DCDCDC;
     color : black;
     z-index : 1;
     width : 394px;
     height: 198px;
     bottom : 125%;
     background-color: white;
     right : -15px;
     padding-right : 10px;
     padding-left : 15px;
     border-radius: 4px;
 }

 .message:before{
    content : '';
    position : absolute;
    top: 100%;
    border-width: 5px; 
    border-style: solid;
    border-color: black transparent transparent transparent;
    left : 90%;
 }

 .message:after{
    content : '';
    position : absolute;
    top: 100%;
    border-width: 4.5px; 
    border-style: solid;
    border-color: white transparent transparent transparent;
    left : 90%;
 }

 .shipping{
     font-weight: bold;
     font-size: 17px;
 }

 .moreshipping{
     margin-top : 10px;
     margin-bottom : 10px;
 }

a:link{
    text-decoration : none;
    color : #0066c0;
 }


a:visited{
   color : #0066c0;
}

a:hover{
    text-decoration: underline;
    color :  #B12704;
}

.arrow{
    height : 8px;
    width : 12px;
}