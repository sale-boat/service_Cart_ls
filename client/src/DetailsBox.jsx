import React from 'react';
import details from './DetailsBox.css';

class DetailsBox extends React.Component {
  constructor(props) {
    super(props);
    this.showMessage = this.showMessage.bind(this);
    this.hideMessage = this.hideMessage.bind(this);
    this.hideBorder = this.hideBorder.bind(this);
    this.state = {
      clicked: false,
      button: true,
    };
  }

  showMessage() {
    const { clicked } = this.state;
    this.setState({ clicked: true });
    this.setState({ button: true });
    if (clicked === true) {
      this.setState({ button: false });
    }
  }

  hideBorder() {
    this.setState({ button: false });
  }

  hideMessage() {
    this.setState({ clicked: false });
  }

  render() {
    const { button } = this.state;
    const { clicked } = this.state;
    const style1 = {
      border: button ? '1px solid orange' : '1px solid white',
      boxShadow: button ? '0 0 3px #ff8000' : '',
      float: 'right',
    };
    let message;
    if (clicked) {
      message = (
        <div className={details.message} onClick={this.hideBorder}>
          <p>
            {' '}
            <div className={details.shipping}>
              <b>Get free shipping</b>
              <button type="button" className={details.exitButton1} style={style1} onClick={this.hideMessage}><b>X</b></button>
            </div>
            {' '}
            <br />
            <b>Free 5-8 business-day shipping</b>
            {' '}
            within the U.S. when you order $25 of eligible items sold or fulfilled by Amazon.
            <br />
            <div className={details.moreshipping}>Or get 4-5 business-day shipping on this item for $5.99. (Prices may vary for AK and HI.)</div>
            <a href="google.com">Learn more about free shipping</a>
          </p>
        </div>
      );
    }
    return (
      <div className={details.tooltip}>
        <span className={details.details} onClick={this.showMessage}>Details</span>
        <img className={details.arrow} src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonarrow.png" alt="drop down arrow" />
        {message}
      </div>
    );
  }
}

export default DetailsBox;
