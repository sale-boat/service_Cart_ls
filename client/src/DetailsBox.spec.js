import React from 'react';
import { mount, shallow } from 'enzyme';
import DetailsBox from './DetailsBox.jsx';

describe('Details Component', () => {
  it('should render', () => {
    const details = shallow(<DetailsBox />);
    expect(details).toMatchSnapshot();
  });

  it('should call show message on click', () => {
    const showMessage = jest.spyOn(DetailsBox.prototype, 'showMessage');
    const wrapper = mount(<DetailsBox /> );
    const detailsText = wrapper.find('.details');
    detailsText.simulate('click');
    expect(showMessage).toHaveBeenCalledTimes(1);
    expect(wrapper.find('.message').exists()).toEqual(true);
  })

  it('should set the clicked state to true when details is clicked', () => {
    const wrapper = shallow(<DetailsBox />);
    const detailsText = wrapper.find('.details');
    detailsText.simulate('click');
    expect(wrapper.state().clicked).toEqual(true);
  })

});
