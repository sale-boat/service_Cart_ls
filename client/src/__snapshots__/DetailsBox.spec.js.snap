// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details Component should render 1`] = `
<div
  className="tooltip"
>
  <span
    className="details"
    onClick={[Function]}
  >
    Details
  </span>
  <img
    alt="drop down arrow"
    className="arrow"
    src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonarrow.png"
  />
</div>
`;
